llvm/lib/Analysis/InlineCost.cpp

```diff
@@ -777,6 +786,42 @@ class InlineCostCallAnalyzer final : public CallAnalyzer {
 
     addCost(SwitchCost);
   }
+
+  // Parses the inline assembly argument to account for its cost. Inline
+  // assembly instructions incur higher costs for inlining since they cannot be
+  // analyzed and optimized.
+  void onInlineAsm(InlineAsm &Arg) override {
+    SmallVector<StringRef, 4> Fragments;
+    Arg.getAsmString().split(Fragments, "\n");
+    int SectionLevel = 0;
+    int InlineAsmInstrCount = 0;
+    for (const auto &Fragment : Fragments) {
```

Comment:
For simple types, I would spell out the type to help the reader:

```suggestion
    for (StringRef Fragment : Fragments) {
```